src/hooks/useFormField.js

```diff
@@ -1,27 +1,52 @@
 import { useState } from "react";
-/*
-A custom hook controlling form field logic.
-It could be extended with:
-    - validation support (a function with validation rules could be passed as an argument)
-    - error message states 
-*/
+import { validateMultipleRules } from "../utils/ValidateRule/ValidateRule";
 
-export default function useFormField(initialState) {
-  const [currentState, setState] = useState(initialState);
+/**
+ * Hook to handle fields and validate them.
+ *
+ * @param {string} initialState - The intial value for the input field.
+ * @param {InputValidator} InputValidator - ValidatorRule that is used to validate the field.
+ *
+ */
 
-  function handleInputChange(e) {
-    switch (e.target.type) {
-      case "checkbox":
-        setState(e.target.checked);
-        break;
-      default:
-        setState(e.target.value);
-    }
-  }
+export default function useFormField(initialState, InputValidator) {
+  const NO_INPUT_VALIDATION = "Input validation is undefinied.";
+
+  const [currentState, setState] = useState(initialState);
+  const [inputValidation, setInputValidation] = useState(InputValidator);
+  const [isInputValid, setIsInputValid] = useState(true);
+  const [errorMessage, setErrorMessage] = useState(NO_INPUT_VALIDATION);
```

Comment:
if this error is what the user sees, then the above message is not something that they should ever see. What are they going to do about it. A default error message could be something like: `Input incorrect`. Or even not say anything and just let the red coloring do the job. Do you see what I mean?

Reply:
I gave some thought and indeed I agree with this. Now it will be valid by default, if there is no validator rule. When checking if it's valid and there is still no validation rule, the value is considered valid.